src/app/pages/history/history.component.ts

```diff
@@ -19,18 +21,35 @@ export class HistoryComponent implements OnInit, OnDestroy {
   private preDataChart: IChart[] = [];
   private sub: Subscription = new Subscription();
 
-  constructor(private historyService: HistoryService) {
-  }
+  constructor(
+    private historyService: HistoryService,
+    public dialog: MatDialog) { }
 
   public ngOnInit(): void {
     this.getDataCategoryAndEvents();
-    this.dataSource = new MatTableDataSource(this.dataTable);
   }
 
   public ngOnDestroy(): void {
     this.sub.unsubscribe();
   }
 
+  public openDialog(): void {
+    const dialogRef = this.dialog.open(AddEventComponent, {
+      data: this.categories
+    });
+    this.sub.add(
+      dialogRef.afterClosed()
+        .subscribe(() => {
+          this.show = false;
+          this.dataTable = [];
+          this.dataChart = [];
+          this.preDataChart = [];
+          this.categories = [];
+          this.getDataCategoryAndEvents();
```

Comment:
what is this? 
`this.show = false;
          this.dataTable = [];
          this.dataChart = [];
          this.preDataChart = [];
          this.categories = [];` - it's not good practise